tests/collections/testsectionlist.ts

```diff
@@ -0,0 +1,110 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2015, Phosphor Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+module tests {
+
+  import SectionList = phosphor.collections.SectionList;
+
+
+
+describe('phosphor.collections - sectionlist', () => {
+
+  describe('insert()', () => {
+
+    it('should return true on success', () => {
+      var obj = new SectionList();
+      expect(obj.size).to.be(0);
+      expect(obj.count).to.be(0);
+      obj.insert(0, 100, 10);
+      expect(obj.size).to.be(1000);
+      expect(obj.count).to.be(100);
+    });
+
+    it('should create heterogenous list', () => {
+      var obj = new SectionList();
+      obj.insert(0, 100, 10);
+      obj.insert(10, 100, 20);
+      expect(obj.size).to.be(3000);
+      expect(obj.count).to.be(200);
+      expect(obj.sizeOf(0)).to.be(10);
+      expect(obj.sizeOf(10)).to.be(20);
+      expect(obj.sizeOf(199)).to.be(10);
+      expect(obj.offsetOf(20)).to.be(300);
+    });
+
+    it('should create another heterogenous list', () => {
+      var obj = new SectionList();
+      obj.insert(0, 75, 10);
+      obj.insert(10, 75, 20);
+      obj.insert(100, 100, 30);
+      obj.insert(150, 150, 40);
+      obj.insert(0, 100, 5);
+      obj.insert(500, 50, 25);
+      expect(obj.size).to.be(13000);
+      expect(obj.count).to.be(550);
+      expect(obj.sizeOf(0)).to.be(5);
+      expect(obj.sizeOf(549)).to.be(25);
+      expect(obj.sizeOf(125)).to.be(20);
+      expect(obj.sizeOf(225)).to.be(30);
+      expect(obj.sizeOf(300)).to.be(40);
+      expect(obj.sizeOf(475)).to.be(10);
+    });
+
+    it('should insert at 0', () => {
```

Comment:
'should insert at `list.count` when index is `>= list.count`', or a more general test which tests all of the parameter conditions described in the API docs.